dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs

```diff
@@ -21,52 +21,44 @@ public sealed class FunctionCallingStepwisePlanner
     /// <summary>
     /// Initialize a new instance of the <see cref="FunctionCallingStepwisePlanner"/> class.
     /// </summary>
-    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
     /// <param name="config">The planner configuration.</param>
     public FunctionCallingStepwisePlanner(
-        Kernel kernel,
         FunctionCallingStepwisePlannerConfig? config = null)
     {
-        Verify.NotNull(kernel);
-        this._kernel = kernel;
-        this._chatCompletion = kernel.GetService<IChatCompletion>();
-
-        ILoggerFactory loggerFactory = kernel.LoggerFactory;
-
-        // Initialize prompt renderer
-        this._promptTemplateFactory = new KernelPromptTemplateFactory(loggerFactory);
-
-        // Set up Config with default values and excluded plugins
         this.Config = config ?? new();
-        this.Config.ExcludedPlugins.Add(RestrictedPluginName);
-
-        this._initialPlanPrompt = this.Config.GetPromptTemplate?.Invoke() ?? EmbeddedResource.Read("Stepwise.InitialPlanPrompt.txt");
+        this._generatePlanYaml = this.Config.GetPromptTemplate?.Invoke() ?? EmbeddedResource.Read("Stepwise.GeneratePlan.yaml");
         this._stepPrompt = this.Config.GetStepPromptTemplate?.Invoke() ?? EmbeddedResource.Read("Stepwise.StepPrompt.txt");
-
-        // Create context and logger
-        this._logger = loggerFactory.CreateLogger(this.GetType());
+        this.Config.ExcludedPlugins.Add(RestrictedPluginName);
     }
 
     /// <summary>
     /// Execute a plan
     /// </summary>
     /// <param name="question">The question to answer</param>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <returns>Result containing the model's response message and chat history.</returns>
     public async Task<FunctionCallingStepwisePlannerResult> ExecuteAsync(
         string question,
+        Kernel kernel,
         CancellationToken cancellationToken = default)
     {
         Verify.NotNullOrWhiteSpace(question);
+        Verify.NotNull(kernel);
+        IChatCompletion chatCompletion = kernel.GetService<IChatCompletion>();
+        ILoggerFactory loggerFactory = kernel.LoggerFactory;
+        ILogger logger = loggerFactory.CreateLogger(this.GetType());
+        var promptTemplateFactory = new KernelPromptTemplateFactory(loggerFactory);
+        var stepExecutionSettings = this.Config.ExecutionSettings ?? new OpenAIPromptExecutionSettings();
 
-        // Add the final answer function
-        this._kernel.ImportPluginFromObject<UserInteraction>();
+        // Clone the kernel and modify it to add the final answer function
+        var clonedKernel = kernel.Clone();
```

Comment:
Why does the kernel need to be cloned? Can you add that detail in the comment i.e., `// Clone kernel to avoid override arguments`